python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -511,18 +529,25 @@ MlirModule synthesizeKernel(const std::string &name, MlirModule module,
       getEnvBool("CUDAQ_MLIR_PRINT_EACH_PASS", false);
 
   PassManager pm(context);
-  pm.addPass(cudaq::opt::createQuakeSynthesizer(name, rawArgs, 0, true));
-  pm.addPass(createCanonicalizerPass());
+  auto kernelNameInQuake = cudaq::runtime::cudaqGenPrefixName + name;
+  auto funcOp = cloned.lookupSymbol<func::FuncOp>(kernelNameInQuake);
+  if (!funcOp)
+    cloned.emitOpError("The kernel '" + name +
+                       "' was not found in the module.");
 
-  // Run state preparation for quantum devices only.
-  // Simulators have direct implementation of state initialization
-  // in their runtime.
   auto &platform = cudaq::get_platform();
-  if (!platform.is_simulator() || platform.is_emulated()) {
-    pm.addPass(cudaq::opt::createConstPropComplex());
-    pm.addPass(cudaq::opt::createLiftArrayAlloc());
-    pm.addPass(cudaq::opt::createStatePreparation());
-  }
+  auto isRemoteSimulator =
+      platform.get_remote_capabilities().serializedCodeExec;
```

Comment:
NVQC will have this set to `false`, but I believe you would want `isRemoteSimulator` set to true in that case, no?

Reply:
Yes, I would need `isRemoteSimulator` to be true for nvqc as well, is there a better way to detect remote simulators?

Reply:
I'm open to other ideas, but perhaps it would be better to add a new struct field to `RemoteCapabilities` in `runtime/cudaq/remote_capabilities.h`. You can then update `getRemoteCapabilities()` in `runtime/common/BaseRestRemoteClient.h` to set it to true similar to how we set `vqe` and `stateOverlap`?

That would be functionally equivalent to simply using the `vqe` or `stateOverlap` field, but it would be a bit more explicit.

Reply:
Added an `isRemoteSimulator` field in the remote capabilities, which I set to true for remote-mqpu and nvqc